bin/plot_river.py

```diff
@@ -0,0 +1,157 @@
+        quit()
+
+    units = "(mgC m$^{-2}$ s$^{-1}$), nutrient converted to C"
+    if version == "new":
+        if tracer == "nitrate":
+            name = "ECO_no3"
+        elif tracer == "phosphate":
+            name = "ECO_pho"
+        elif tracer == "silicate":
+            name = "ECO_sil"
+        else:
+            name = "rivers"
+            units = "(m s$^{-1}$)"
+
+    if version == "old":
+        if tracer == "nitrate":
+            name = "rivnitr"
+        elif tracer == "phosphate":
+            name = "rivphos"
+        elif tracer == "silicate":
+            name = "rivsili"
+        else:
+            name = "rivers"
+            units = "(m s$^{-1}$)"
+
+    if region == "TP0":
+        region = "TP0a1.00"
+    if region == "TP2":
+        region = "TP2a0.10"
+    if region == "TP4":
+        region = "TP4a0.12"
+    if region == "TP5":
+        region = "TP5a0.06"
+    if region == "NAT":
+        region = "NATa1.00"
+    if region == "NA2":
+        region = "NA2a0.80"
+
+
+    abgrid = abfile.ABFileGrid(workdir + user + "/" + \
+        region + "/topo/regional.grid","r")
+    plon=abgrid.read_field("plon")
+    plat=abgrid.read_field("plat")
+    jdm,idm=plon.shape
+
+    abriver = abfile.AFile(idm,jdm,workdir + user + "/" + \
+        region + "/force/rivers/" + experiment + \
+            "/" + name + ".a","r")
+    river  = abriver.read_record(np.int(month)-1)
+    abriver.close()
+
+    if version == "old":
+        if tracer == "nitrate":
+            river = river * 12.01 * 6.625 / 14.007
+        elif tracer == "phosphate":
+            river = river * 12.01 * 106. / 30.97
+        elif tracer == "silicate":
+            river = river * 12.01 * 106. / 28.09
+
+    abdepth = abfile.ABFileBathy(workdir + user + "/" + \
+        region + "/force/rivers/SCRATCH/regional.depth.b","r",idm=idm,jdm=jdm)
+    depthm=abdepth.read_field("depth")
+    river = np.ma.masked_where(depthm<10.,river)
+
+    # now plot
+    fig=plt.figure(figsize=(6,5),facecolor='w')
+    ax = fig.add_subplot(1,1,1)
+    ax.set_position([0.01,0.02,0.865,0.96])
+    cmap = plt.get_cmap(cmap)
+    ax.set_facecolor('xkcd:gray')
+    pmesh = plt.pcolormesh(river,cmap=cmap)
+    cb=ax.figure.colorbar(pmesh)
+    if clim is not None : pmesh.set_clim(clim)
+    plt.text(0.015,1.05, "%s %s %s %s" %("river", tracer, "flux month:",\
+        month.zfill(2)),transform=ax.transAxes,FontSize=13)
+    plt.text(0.015,1.005, units,transform=ax.transAxes,FontSize=8)
+
+    # save figure
+    fig.canvas.print_figure(workdir + user + "/" + \
+        region + "/force/rivers/" + experiment + \
+            "/" + name + ".png",dpi=180)
+
+if __name__ == "__main__" :
+
+    class ClimParseAction(argparse.Action) :
+        def __call__(self, parser, args, values, option_string=None):
+           tmp = values.split(",")
+           tmp = [float(elem) for elem in tmp[0:2]]
+           setattr(args, self.dest, tmp)
+
```

Comment:
This works great, I propose something like this to make it slightly more user friendly:

    print("Figure saved in " + workdir + user + "/" + \
        region + "/force/rivers/" + experiment + \
            "/" + name + ".png")